modin/pandas/groupby.py

```diff
@@ -846,7 +846,7 @@ def fillna(self, *args, **kwargs):
             squeeze=self._squeeze,
             **new_groupby_kwargs,
         )
-        return work_object._check_index(
+        return work_object._check_index_name(
```

Comment:
I see this is reverting 6c82427800acb3ddd7105f97a8abc0eca8699a90. That's fine by me.

Reply:
Yes I reverted because `_check_index_name` uses QC's `set_index` method, which I thought is safer than `result.set_index` of `_check_index` for chained operations.